externals/llvm-external-projects/torch-mlir-dialects/include/torch-mlir-dialects/Dialect/Tcp/IR/TcpInterfaces.td

```diff
@@ -0,0 +1,30 @@
+#ifndef TORCH_MLIR_DIALECT_TCP_INTERFACES
+#define TORCH_MLIR_DIALECT_TCP_INTERFACES
+
+include "torch-mlir-dialects/Dialect/Tcp/IR/TcpBase.td"
+
+def TcpOpInterface : OpInterface<"TcpOpInterface"> {
+  let description = [{
+    Implemented by all ops in TCP.
+  }];
+}
+
+def TcpUnaryOpInterface : OpInterface<"TcpUnaryOpInterface"> {
+  let description = [{
+    Implemented by all unary ops in TCP.
+  }];
+}
+
+def TcpBinaryOpInterface : OpInterface<"TcpBinaryOpInterface"> {
+  let description = [{
+    Implemented by all binary ops in TCP.
+  }];
+}
+
+def TcpTernaryOpInterface : OpInterface<"TcpTernaryOpInterface"> {
+  let description = [{
+    Implemented by all ternary ops in TCP.
+  }];
+}
```

Comment:
Couple of points:

1. Please document the broadcasting semantics here.
2. The description itself is circular -- IMO this is a good place to define what "binary op" means.
3. Should this op interface define operations like `lhs` and `rhs` for binary ops?
4. Should this be a trait instead?  If not, I suggest adding a oneliner stating why not.

Reply:
Good point regarding trait vs interface. I'll spend some more time trying to figure out what works best for our usecase here. For now, I'm removing all interfaces.